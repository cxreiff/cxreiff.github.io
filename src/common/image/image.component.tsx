import { useState } from "react";

type ImageProps = {
  aspect: number;
  source: string;
  alt?: string;
  className?: string;
};

import { Frame } from "src/common/frame/frame.component";

export const Image = ({ aspect, source, alt, className }: ImageProps) => {
  const [loading, setLoading] = useState(true);

  return (
    <Frame aspect={aspect} loading={loading} className={className}>
      <img src={source} onLoad={() => setLoading(false)} alt={alt} />
    </Frame>
  );
};
