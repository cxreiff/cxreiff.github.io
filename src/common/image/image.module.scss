@use '../../shared.scss' as *;

.image {
  position: relative;
  width: 100%;
  height: 100%;

  &>svg {
    position: relative;
  }

  &>img {
    max-width: 100%;
    max-height: 100%;
    position: absolute;
    z-index: 1;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    margin: auto;
  }

  &.hidden>img {
    display: none;
  }

  &:not(.hidden)>img {
    animation: fadeIn ease-in 0.4s;
  }
}

@keyframes fadeIn {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}
