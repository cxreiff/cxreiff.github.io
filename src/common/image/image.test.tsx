import { mount, ReactWrapper } from 'enzyme'

import { Image } from './image.component'

describe('imageLoader', () => {
    
    let component: ReactWrapper<typeof Image, void>
    
    beforeEach(() => {
        component = mount<typeof Image, void>(
            <Image
                aspect={1/1}
                source={'test'}
                alt={'test'}
                className={'test'}
            />
        )
    })

    afterEach(() => {
        component.unmount()
    })
    
    it('should render', () => {
        expect(component).toMatchSnapshot()
    })

    it('should reveal image on load', () => {
        component.find('img').simulate('load')
        expect(component).toMatchSnapshot()
    })
})
