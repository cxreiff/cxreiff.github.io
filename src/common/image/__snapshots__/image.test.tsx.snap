// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div>
  <div
    class="frame hidden"
  >
    <img
      alt="test image"
      src="test"
    />
    <svg
      viewBox="0 0 1.3333333333333333 1"
    />
  </div>
</div>
`;

exports[`should unhide on image load 1`] = `
<div>
  <div
    class="frame"
  >
    <img
      alt="test image"
      src="test"
    />
    <svg
      viewBox="0 0 1.3333333333333333 1"
    />
  </div>
</div>
`;
